src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -41,7 +40,7 @@ public MessagePump(ConnectionConfiguration configuration, IAmazonS3 s3Client, IA
                 // in that time.
                 try
                 {
-                    await sqsClient.PurgeQueueAsync(queueUrl).ConfigureAwait(false);
+                    await sqsClient.PurgeQueueAsync(queueUrl, CancellationToken.None).ConfigureAwait(false);
```

Comment:
I wonder if we should even support this. I'd rather throw an exception here so that users are not bitten. I mean, people would rightfully expect that the `PurgeOnStartup` will clean up the queue before you start sending stuff. But with SQS you are ensured to lose messages for up to 1 minute after such an endpoint is started.

Reply:
Actually this could be a separate issue...

Reply:
I agree with @SzymonPobiega, purge can lead to bad things

Reply:
Agreed

Note: Core have a few acceptance tests that relies on the purge to be immediate cc @timbussmann

Reply:
Agreed but not a concern for this PR. There is also a platdev issue which
talks about removing entirely from the core